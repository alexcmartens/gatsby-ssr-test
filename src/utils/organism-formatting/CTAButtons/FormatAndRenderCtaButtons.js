import React from "react"
import renderPaddingOrFullBleed from "../renderPaddingOrFullBleed"
import { Link } from "gatsby"
import { CTAButtons } from "terra-one"

const FormatAndRenderCtaButtons = ({ data }) => {
  return (
    <>
      {renderPaddingOrFullBleed(
        <CTAButtons
          variant={data?.variant}
          alignment={data?.alignment}
          buttons={data?.ctaButtons}
          Link={Link}
        />
      )}
    </>
  )
}

export default FormatAndRenderCtaButtons
