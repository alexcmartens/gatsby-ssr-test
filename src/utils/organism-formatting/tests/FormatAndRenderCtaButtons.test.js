import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderCtaButtons from "../CTAButtons/FormatAndRenderCtaButtons"

describe("FormatAndRenderCtaButtons", () => {
  it("renders correctly", () => {
    const data = {
      variant: "dark",
      alignment: "left",
      ctaButtons: [
        {
          text: "123",
          url: "/",
        },
        {
          text: "123",
          url: "/",
        },
      ],
    }

    const tree = renderer
      .create(<FormatAndRenderCtaButtons data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
