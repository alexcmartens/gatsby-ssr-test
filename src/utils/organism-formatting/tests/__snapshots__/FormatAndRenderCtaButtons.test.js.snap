// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderCtaButtons renders correctly 1`] = `
<section
  className="default-layout   mockconstructor
      "
  style={
    Object {
      "backgroundColor": undefined,
      "marginBottom": "",
    }
  }
>
  <section
    className="default-layout__grid 
          
        "
    style={
      Object {
        "backgroundColor": "",
        "marginBottom": "",
      }
    }
  >
    <div
      className="default-layout__children "
    >
      <div
        alignment="left"
        buttons={
          Array [
            Object {
              "text": "123",
              "url": "/",
            },
            Object {
              "text": "123",
              "url": "/",
            },
          ]
        }
        variant="dark"
      />
    </div>
  </section>
</section>
`;
